integrations/airflow/marquez_airflow/dag.py

```diff
@@ -77,44 +75,47 @@ def create_dagrun(self, *args, **kwargs):
 
         return dagrun
 
-    def _register_dagrun(self, dagrun, execution_date, run_args):
-        self.log.debug(f"self.task_dict: {self.task_dict}")
+    # TODO: do we actually need to start tasks here or just register a hierarchy?
+    # Doing it other way would require to hook up to
+    # scheduler, where tasks are actually started
+    def _register_dagrun(self, dagrun, execution_date):
+        self.log.error(f"---------------------------self.task_dict: {self.task_dict}")
         # Register each task in the DAG
         for task_id, task in self.task_dict.items():
             t = self._now_ms()
             try:
-                steps = self._extract_metadata(dagrun, task)
-                [_MARQUEZ.create_job(
-                    step, self._get_location(task), self.description)
-                    for step in steps]
-                marquez_jobrun_ids = [_MARQUEZ.create_run(
-                    self.new_run_id(),
-                    step,
-                    run_args,
+                step = self._extract_metadata(dagrun, task)
+
+                job_name = self._marquez_job_name(self.dag_id, task.task_id)
+                task_run_id = _MARQUEZ.start_task(
+                    job_name,
+                    self.description,
+                    DagUtils.to_iso_8601(self._now_ms()),
```

Comment:
ideally this is the official start time per airflow.

Reply:
Good catch, we can use task instance's start time here.